packages/autorest.python/autorest/codegen/models/enum_type.py

```diff
@@ -178,7 +178,7 @@ def type_annotation(self, **kwargs: Any) -> str:
             if not kwargs.get("is_operation_file", False):
                 model_name = f'"{model_name}"'
 
-            return f"Union[{self.value_type.type_annotation(**kwargs)}, {model_name}]"
+            return f"Union[{model_name}, {self.value_type.type_annotation(**kwargs)}]"
```

Comment:
why does changing the ordering of the union fix the issue? I don't think this is a good way to fix the issue because it's brittle and looking at this ordering, no one would think if they switched the ordering later on, enum serialization would be broken. I'm assuming the actual issue is something wrong in our model base serialization. We should be able to fix the model base serialization so we can maintain the same ordering

Reply:
type annotation of enum is Union and Python SDK will try to deserialize it with element by order: https://github.com/Azure/azure-sdk-for-python/blob/1f88d675782020eb7e092f94fb58a3d8130d5ccf/sdk/anomalydetector/azure-ai-anomalydetector/azure/ai/anomalydetector/_model_base.py#L491-L499.
Now "str" is first place so deserialized type is always str for enum

Reply:
We should special case this for a union with enum and the base type in it. The order of types in `Union` should not affect how we deserialize a type

Reply:
It should be a bug for desterilization logic of dpg models.

Reply:
`The order of types in Union should not affect how we deserialize a type`
- When deserialize `X: Union[str, int]` with `"123"`, result is string `"123"`
- When deserialize `X: Union[int, str]` with `"123"`, result is int `123`
So actually the order of types will influence result. For enum, when users call `.enum_property`, SDK shall return enum type so SDK shall permit enum type annotation is in first place.

Reply:
After discussion with yuchao offline, I think it's better to adjust the union typing sequence for enum in Python. I believe we should deserialize the payload according to the user defined sequence of union types, instead of guessing which type should try first. For example, we have a payload `"2"` with type `str|int` should be deserialized to a string `"2"`, while with type `int|str` should be an int `2`.